spring-security-karaf-login-module/src/test/java/org/pentaho/platform/osgi/auth/spring/SpringSecurityLoginModuleTest.java

```diff
@@ -196,8 +212,15 @@ public void testExceptions() throws Exception {
     loginModule.commit();
 
     verify( authenticationManager ).authenticate( argThat( new AuthenticationManagerMatcher( "joe" ) ) );
-    assertEquals( 4, subject.getPrincipals().size() );
-    subject.getPrincipals().toArray()[ 3 ].equals( "karaf_admin" );
+    assertEquals( 10, subject.getPrincipals().size() );
+    assertTrue( containsPrincipalName( subject, "karaf_admin" ) );
+    assertTrue( containsPrincipalName( subject, "karaf_admin" ) );
```

Comment:
Repeated assertion.

Reply:
no longer an issue, "karaf_admin" is no longer a role in the code.